### PR TITLE
NO-ISSUE: Add bonding support to assisted test infra

### PR DESCRIPTION
**Pull request by ori-amizur**
Bonding is available for libvirt only.
When bonding is enabled, 2 bonded interfaces are bonded, one per network.
3 additional environment variables:
IS_BONDED(boolean): Should the interfaces be bonded.  Default is false. NUM_BONDED_SLAVES(int): The number of slave interfaces per bond. Default is 2.
BONDING_MODE(string): The bonding mode. Default is active-backup.

/cc @filanov 
/cc @eliorerz

### CODE REVIEW

**Review by osherdp — COMMENTED** [review]
/cc @adriengentil

File: src/assisted_test_infra/test_infra/controllers/node_controllers/terraform_controller.py [hunk 1]
```diff
@@ -137,6 +137,10 @@ def _get_specific_tf_entity_params(self) -> Dict[str, str]:
     def list_nodes(self) -> List[Node]:
         return self.list_nodes_with_name_filter(self._entity_name.get())
 
+    def generate_macs(self, count: int) -> List[str]:
```
[the remark this answers is not in the document]
Comment (ori-amizur) [reply src/assisted_test_infra/test_infra/controllers/node_controllers/terraform_controller.py]: OK

File: src/assisted_test_infra/test_infra/tools/static_network.py [hunk 2]
```diff
@@ -105,12 +149,45 @@ def _prepare_host_static_network_data(
...
 
 
+def _prepare_host_bonded_static_network_data(
+    primary_macs: List[str],
+    secondary_macs: List[str],
+    machine_cidrs: List[str],
+    provisioning_cidrs: List[str],
+    count: int,
+    num_bonded_slaves,
+    bonding_mode,
```
[the remark this answers is not in the document]
Comment (ori-amizur) [reply src/assisted_test_infra/test_infra/tools/static_network.py]: OK

File: src/assisted_test_infra/test_infra/tools/static_network.py [hunk 3]
```diff
@@ -105,12 +149,45 @@ def _prepare_host_static_network_data(
...
 
+def _prepare_host_bonded_static_network_data(
+    primary_macs: List[str],
+    secondary_macs: List[str],
+    machine_cidrs: List[str],
+    provisioning_cidrs: List[str],
+    count: int,
+    num_bonded_slaves: int,
+    bonding_mode: str,
+) -> Dict[str, List[dict]]:
```
Comment (adriengentil) [inline src/assisted_test_infra/test_infra/tools/static_network.py:160]:
```suggestion
) -> Dict[str, List[Dict[str, Any]]]:
```

File: src/assisted_test_infra/test_infra/tools/static_network.py [hunk 4]
```diff
@@ -134,6 +211,41 @@ def _prepare_interface(logical_interface: str, cidrs: List[str], host_count: int
...
 
 
+def _prepare_bonded_interface(
+    logical_interface: str,
+    starting_physical_interface: int,
+    cidrs: List[str],
+    host_count: int,
+    num_bonded_slaves: int,
+    bonding_mode: str,
+) -> Dict[str, str]:
```
Comment (adriengentil) [inline src/assisted_test_infra/test_infra/tools/static_network.py:221]:
```suggestion
) -> Dict[str, Any]:
```
the call to `_prepare_link_aggregation` below stores a dict as a value

File: src/assisted_test_infra/test_infra/tools/static_network.py [hunk 5]
```diff
@@ -105,12 +149,45 @@ def _prepare_host_static_network_data(
...
+
 def _prepare_interfaces(machine_cidrs: List[str], provisioning_cidrs: List[str], count: int) -> List[Dict[str, str]]:
     primary_interface = _prepare_interface(_PRIMARY_LOGICAL_INTERFACE, machine_cidrs, count)
     secondary_interface = _prepare_interface(_SECONDARY_LOGICAL_INTERFACE, provisioning_cidrs, count)
     return [primary_interface, secondary_interface]
 
 
+def _prepare_bonded_interfaces(
+    machine_cidrs: List[str], provisioning_cidrs: List[str], count: int, num_bonded_slaves: int, bonding_mode: str
+) -> List[Dict[str, str]]:
```
Comment (adriengentil) [inline src/assisted_test_infra/test_infra/tools/static_network.py:183]:
```suggestion
) -> List[Dict[str, Any]]:
```

**Review by adriengentil — COMMENTED** [review]
Few nits, otherwise it looks good to me

File: src/assisted_test_infra/test_infra/tools/static_network.py (hunk 3 above)
Comment (ori-amizur) [reply src/assisted_test_infra/test_infra/tools/static_network.py:160]: OK

File: src/assisted_test_infra/test_infra/tools/static_network.py [hunk 6]
```diff
@@ -73,6 +70,53 @@ def generate_static_network_data_from_tf(tf_folder: str) -> List[Dict[str, List[
     return network_config
 
 
+def _generate_bonded_static_network_data_from_tf(
+    tfvars: Dict[str, any], num_bonded_slaves: int, bonding_mode: str
```
[the remark this answers is not in the document]
Comment (ori-amizur) [reply src/assisted_test_infra/test_infra/tools/static_network.py:74]: OK

File: src/assisted_test_infra/test_infra/tools/static_network.py (hunk 4 above)
Comment (ori-amizur) [reply src/assisted_test_infra/test_infra/tools/static_network.py:221]: OK

File: src/assisted_test_infra/test_infra/tools/static_network.py (hunk 5 above)
Comment (ori-amizur) [reply src/assisted_test_infra/test_infra/tools/static_network.py]: OK

**Review by adriengentil — APPROVED** [review]

**Pull request merged**